src/main/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApproval.java

```diff
@@ -536,6 +533,40 @@ public ScriptApproval() {
         if (changed) {
             save();
         }
+        // only call on subsequent load to avoid cycle
+        if (Jenkins.get().getInitLevel() == InitMilestone.COMPLETED) {
+            try {
+                LOG.log(Level.FINE, "Reconfiguring ScriptApproval after loading configuration from disk");
+                reconfigure();
```

Comment:
Looks like this is what is causing issues with the tests, most specifically with the hot plugin load.